shared-bindings/gamepad/__init__.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 Radomir Dopieralski for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include "py/obj.h"
+#include "py/runtime.h"
+#include "py/mphal.h"
+#include "shared-module/gamepad/__init__.h"
+
+
+gamepad_obj_t* gamepad_singleton = NULL;
+
+
+
+//| :mod:`gamepad` --- Button handling
+//  ==================================
+//|
+//| .. module:: gamepad
+//|   :synopsis: Button handling
+//|   :platform: SAMD21
+//|
+
+//| ..method:: GamePad.get_pressed()
+//|
+//| Get the status of gamepad pressed since the last call.
+//|
+//| Returns an 8-bit number, with bits that correspond to gamepad, which
+//| have been pressed (or held down) since the last call to this function
+//| set to 1, and the remaining bits set to 0.
+//|
+STATIC mp_obj_t gamepad_get_pressed(mp_obj_t self_in) {
+    gamepad_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_obj_t gamepad = MP_OBJ_NEW_SMALL_INT(self->pressed);
+    self->pressed = 0;
+    return gamepad;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(gamepad_get_pressed_obj, gamepad_get_pressed);
+
+STATIC mp_obj_t gamepad_make_new(const mp_obj_type_t *type, size_t n_args,
+        size_t n_kw, const mp_obj_t *args);
+STATIC const mp_rom_map_elem_t gamepad_locals_dict_table[] = {
+    { MP_OBJ_NEW_QSTR(MP_QSTR_get_pressed),  MP_ROM_PTR(&gamepad_get_pressed_obj)},
+};
+STATIC MP_DEFINE_CONST_DICT(gamepad_locals_dict, gamepad_locals_dict_table);
+const mp_obj_type_t gamepad_type = {
+    { &mp_type_type },
+    .name = MP_QSTR_GamePad,
+    .make_new = gamepad_make_new,
+    .locals_dict = (mp_obj_dict_t*)&gamepad_locals_dict,
+};
```

Comment:
Please move the GamePad object and header to a separate file to match TouchIn and others. https://github.com/adafruit/circuitpython/blob/master/shared-bindings/touchio/TouchIn.c#L54

Reply:
Done.